### PR TITLE
Added GoogleDocStrings & Added TypeHint.

### PR DESCRIPTION
**Pull request by ranbir7**
Shifted Some import statements
and adjusted indentation of GoogleDocStrings.

### CODE REVIEW

**Comment by ranbir7** [comment]
Is this not correct?? 
@ConnorStoneAstro

File: autoprof/fit/gradient.py [hunk 1]
```diff
@@ -36,15 +36,15 @@ class Grad(BaseOptimizer):
     """
 
     
-    def __init__(self, model, initial_state = None, **kwargs):
+    def __init__(self, model: object, initial_state = None, **kwargs) -> None:
```
Comment (ConnorStoneAstro) [inline autoprof/fit/gradient.py]: The `model` argument should be an `AutoProf_Model` object. Just like below you can use the string "AutoProf_Model"

File: autoprof/fit/gradient.py [hunk 2]
```diff
@@ -87,13 +87,13 @@ def step(self):
             AP_config.ap_logger.info(f"gradient: {self.current_state.grad}")
         self.optimizer.step()
         
-    def fit(self):
+    def fit(self) -> None:
```
Comment (ConnorStoneAstro) [inline autoprof/fit/gradient.py]: The `fit` function returns `self` not `None`

File: autoprof/fit/iterative.py [hunk 3]
```diff
@@ -1,11 +1,19 @@
...
+#According to PEP8 conventions
+from typing import Dict,Any
 import os
-import torch
+import matplotlib.pyplot as plt
+
 import numpy as np
+import torch
+
 from time import time
```
Comment (ConnorStoneAstro) [inline autoprof/fit/iterative.py]: the `time` library should move up. I think just under `os`

File: autoprof/fit/iterative.py [hunk 4]
```diff
@@ -22,9 +30,23 @@ class Iter(BaseOptimizer):
...
+        method_kwargs: Keyword arguments to pass to `method`.
+        iteration: The number of iterations performed.
+        lambda_history: A list of the states at each iteration step.
+        loss_history: A list of the losses at each iteration step
     """
 
-    def __init__(self, model, method, initial_state = None, max_iter = 100, method_kwargs = {}, **kwargs):
+    def __init__(self, 
+                model: 'AutoProf_Model', 
+                method: callable,
```
Comment (ConnorStoneAstro) [inline autoprof/fit/iterative.py]: The `method` argument should be a `BaseOptimizer` object

File: autoprof/fit/iterative.py [hunk 5]
```diff
@@ -80,7 +111,12 @@ def step(self):
 
         self.iteration += 1
         
-    def fit(self):
+    def fit(self) -> None:
```
Comment (ConnorStoneAstro) [inline autoprof/fit/iterative.py]: `fit` returns `self` not `None`

**Review by ConnorStoneAstro — COMMENTED** [review]
Hi Ranbir! Sorry for the late reply, I often don't work over the weekend. The changes look great, I just added a few comments to be fixed up before I approve the PR.

File: autoprof/fit/gradient.py (hunk 2 above)
Comment (ConnorStoneAstro) [reply autoprof/fit/gradient.py]: Sorry, by returns `self` I should have said it returns a `BaseOptimizer` object. That object just happens to be self

File: autoprof/fit/iterative.py (hunk 5 above)
Comment (ConnorStoneAstro) [reply autoprof/fit/iterative.py]: Same as before by returns `self` I should have said it returns a `BaseOptimizer` object

**Comment by ranbir7** [comment]
Okay I'll be solving each one of them...soon

File: autoprof/fit/iterative.py (hunk 3 above)
Comment (ranbir7) [reply autoprof/fit/iterative.py]: okay I'll resolve it
there's this rule in PEP8 convention where 
if any import
lets take an example
"from .. import AP_config"
should be "import AP_config".
I have only thought about shifting the AP_config file to the required file's current dir.

Comment (ConnorStoneAstro) [reply autoprof/fit/iterative.py]: Unfortunately we can't do that since AP_config is also used in other folders, but just the `fit\` one. The only way to fix this would be to move everything into a single large folder, which would be too messy. For now we'll just have to leave `from .. import AP_config` as is

**Review by ConnorStoneAstro — COMMENTED** [review]
Hi Ranbir, this is looking great! Since this is a code update there are extra checks I need to run, but it all looks good. I will run the checks tomorrow and if all goes well it can be merged!

**Review by ConnorStoneAstro — APPROVED** [review]
Ran my local tests and everything seems to be working properly! Thanks again, these updates are super helpful!
